java/piranha/src/test/java/com/uber/piranha/CorePiranhaTest.java

```diff
@@ -1028,4 +1036,228 @@ public void testRemoveSpecificAPIpatterns() throws IOException {
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+    bcr = addMockAPIToNameSpace(bcr);
+    bcr.addInput("XPMethodChainCases.java").addOutput("XPMethodChainCasesTreatment.java").doTest();
+  }
+
+  String trgtProp = "config/properties-test-rt.json";
+
+  @Test
+  public void testMethodChainControl() throws IOException {
+    String staleFlag = "stale_flag";
+    String isTreated = "false";
+    String srcProp = "src/test/resources/config/properties_method_chain.json";
+    transformAndCreateNewPropertyFile(
+        srcProp, trgtProp, staleFlag, Boolean.parseBoolean(isTreated));
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", staleFlag);
+    b.putFlag("Piranha:IsTreated", isTreated);
+    b.putFlag("Piranha:ArgumentIndexOptional", "true");
+    b.putFlag("Piranha:Config", trgtProp);
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+    bcr = addMockAPIToNameSpace(bcr);
+    bcr.addInput("XPMethodChainCases.java").addOutput("XPMethodChainCasesControl.java").doTest();
+  }
+
+  private BugCheckerRefactoringTestHelper addMockAPIToNameSpace(
+      BugCheckerRefactoringTestHelper bcr) {
+    bcr = bcr.addInput("mock/PVal.java").expectUnchanged();
+    bcr = bcr.addInput("mock/Parameter.java").expectUnchanged();
+    bcr = bcr.addInput("mock/BoolParam.java").expectUnchanged();
+    bcr = bcr.addInput("mock/BoolParameter.java").expectUnchanged();
+    bcr = bcr.addInput("mock/SomeParamRev.java").expectUnchanged();
+    bcr = bcr.addInput("mock/OverlappingNameInterface.java").expectUnchanged();
+    bcr = bcr.addInput("mock/SomeOtherInterface.java").expectUnchanged();
+    bcr = bcr.addInput("mock/StaticMthds.java").expectUnchanged();
+    return bcr;
+  }
+
+  // Deletes the abstract method declaration but does not delete the
+  // chained method usages
+  @Test
+  public void testMethodTestDoNotAllowsChainFlag() throws IOException {
+    String staleFlag = "stale_flag";
+    String isTreated = "false";
+    String srcProp = "src/test/resources/config/properties_method_chain_not_allow_chained.json";
+    transformAndCreateNewPropertyFile(
+        srcProp, trgtProp, staleFlag, Boolean.parseBoolean(isTreated));
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", staleFlag);
+    b.putFlag("Piranha:IsTreated", isTreated);
+    b.putFlag("Piranha:ArgumentIndexOptional", "true");
+    b.putFlag("Piranha:Config", trgtProp);
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+    bcr = addMockAPIToNameSpace(bcr);
+    bcr.addInputLines(
```

Comment:
can we not just use the same input file as before and use a different output file for testing?

Reply:
Yup.